### ISSUE TITLE
Add support for attachments

### ISSUE THREAD

**Issue by RhetTbull** [RhetTbull/macnotesapp#15]
Attachments are not currently handled.

**Comment by RhetTbull** [comment]
@SKaplanOfficial I wasn't aware of PyXa. Had played around with ScriptingBridge a bit on another project but got frustrated by lack of documentation. I just toyed around with PyXA -- amazing work!  I may just rewrite macnotesapp to use PyXA as it is so much faster. I have a similar project [photoscript](https://github.com/RhetTbull/PhotoScript) that uses the same AppleScript techniques that might benefit from PyXA (though photoscript currently supports creating folders/albums, moving photos from one album to another, etc. that don't appear to be supported in PyXA yet).

**Comment by SKaplanOfficial** [comment]
@RhetTbull Yeah, ScriptingBridge is a bit of a pain to work with, but I've found it to be robust enough for my needs. I wanted PyXA to serve as sort of an extended reference for ScriptingBridge/pyobjc as a whole since the existing documentation was... lacking. PyXA currently supports creating folders and albums, but moving photos is something I'm working on (moving AS objects in general, really, since the implementation varies slightly across different apps).

**Comment by SKaplanOfficial** [comment]
> Quick question: is it possible to access an existing album by path (e.g. `Folder1/SubFolder1/AlbumName` and get back an album object?) 

Yes and no; you can't access it by path in the form you specified, but you can use `by_name()` to get an album/folder object quickly. For example:

```python
import PyXA
app = PyXA.Application("Photos")
print(app.folders().by_name("Folder1").folders().by_name("Folder2").albums().by_name("Album1"))
>>> <<class 'PyXA.apps.PhotosApp.XAPhotosAlbum'>Album1, id=F0DA93AB-CC92-23B2-9E16-973F6A062468/L0/040>
```

This should work near-instantly in most cases, unless your `by_` call is to something that has to be queried per-object, e.g. for the body text (not plaintext) of notes.

**Comment by RhetTbull** [comment]
Perfect -- thanks!

**Comment by RhetTbull** [comment]
I'm able to add attachments (on Ventura at least, does not work on Catalina, not sure about Big Sur/Monterey) using the following AppleScript:

```applescript
tell application "Notes"
	set theNote to note id "x-coredata://B36F4BF3-D2F0-4997-BD9B-96A608B6D5E0/IMAPNote/p40"
	set theFile to POSIX file "/Users/rhet/Downloads/debug.txt"
	make new attachment at theNote with data theFile
end tell
```

Can't figure out a way to do this via ScriptingBridge though.

**Comment by SKaplanOfficial** [comment]
This works for me to add an attachment on Ventura:

```python
import AppKit
import ScriptingBridge

app = ScriptingBridge.SBApplication.alloc().initWithBundleIdentifier_("com.apple.notes")
note = app.notes()[0]
url = AppKit.NSURL.alloc().initFileURLWithPath_("/Users/exampleUser/Documents/Example.jpg")
new_attachment = app.classForScriptingClass_("attachment").alloc().initWithData_andProperties_(url, {})
note.attachments().addObject_(new_attachment)
```

But the AppleScript approach is probably good enough (and maybe better on other macOS versions).

**Comment by RhetTbull** [comment]
Thanks! Will give that a try! I've been digging through your PyXA source to figure out ScriptingBridge and its been immensely helpful but it's still a lot of trial and error. I did get most of macnotesapp rewritten to use ScriptingBridge though and it's *much* faster now.  I'd like to contribute back to PyXA once I get through my current round of projects.  (for example, I've got a [partial interface to Photos via PhotoKit](https://github.com/RhetTbull/osxphotos/blob/master/osxphotos/photokit.py) that's much faster than scripting)
